scripts/common_startup.sh

```diff
@@ -42,16 +54,46 @@ if [ ! -f $GALAXY_CONFIG_FILE ]; then
     GALAXY_CONFIG_FILE=config/galaxy.ini.sample
 fi
 
-if [ $FETCH_EGGS -eq 1 ]; then
-    python ./scripts/check_eggs.py -q -c $GALAXY_CONFIG_FILE
-    if [ $? -ne 0 ]; then
-        echo "Some eggs are out of date, attempting to fetch..."
-        python ./scripts/fetch_eggs.py -c $GALAXY_CONFIG_FILE
-        if [ $? -eq 0 ]; then
-            echo "Fetch successful."
+if [ $SET_VENV -eq 1 ]; then
+    # If not already in a virtualenv and .venv does not exist, attempt to create it.
+    if [ -z "$VIRTUAL_ENV" -a ! -d .venv ]
+    then
+        vvers=13.1.0
+        vurl="https://pypi.python.org/packages/source/v/virtualenv/virtualenv-${vvers}.tar.gz"
+        if command -v virtualenv >/dev/null; then
+            virtualenv .venv
         else
-            echo "Fetch failed."
-            exit 1
+            vtmp=`mktemp -d -t galaxy-virtualenv-XXXXXX`
+            vsrc="$vtmp/`basename $vurl`"
+            ( if command -v curl >/dev/null; then
+                curl --silent -o $vsrc $vurl
+            elif command -v wget >/dev/null; then
+                wget --quiet -O $vsrc $vurl
+            else
+                python -c "import urllib; urllib.urlretrieve('$vurl', '$vsrc')"
+            fi ) &&
+            tar zxf $vsrc -C $vtmp &&
+            python $vtmp/virtualenv-$vvers/virtualenv.py .venv &&
+            rm -rf $vtmp
         fi
     fi
-fi	
+
+    # If there is a .venv/ directory, assume it contains a virtualenv that we
+    # should run this instance in.
+    if [ -d .venv ];
+    then
+        printf "Activating virtualenv at %s/.venv\n" $(pwd)
+        . .venv/bin/activate
+    fi
+fi
+
+if [ -z "$VIRTUAL_ENV" ]; then
+    echo "ERROR: A virtualenv cannot be found or created. Please create a virtualenv in .venv, or activate one."
+    exit 1
+fi
+
+
+if [ $FETCH_WHEELS -eq 1 ]; then
+    pip install git+https://github.com/natefoo/pip@linux-wheels
```

Comment:
This currently checks out the repository again every time I run it. Changing to `pip install -e git+https://github.com/natefoo/pip@linux-wheels#egg=pip` allows it just to update.